modules/auth/oauth2/oauth2.go

```diff
@@ -0,0 +1,132 @@
+package oauth2
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/gorilla/sessions"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/github"
+	"gopkg.in/macaron.v1"
+	"net/http"
+	"os"
+	"github.com/go-macaron/session"
+	"github.com/satori/go.uuid"
+	"encoding/base64"
+	"strings"
+)
+
+var (
+	sessionUsersStoreKey         = "gitea-oauth-sessions"
+	oAuthUserSessionKey          = "oauth2-user"
+	oAuthStateUniqueIDSessionKey = "oauth2-state-key"
+)
+
+func init() {
+	gothic.Store = sessions.NewFilesystemStore(os.TempDir(), []byte(sessionUsersStoreKey))
```

Comment:
Think you mean : setting.SessionConfig.ProviderConfig = "data/sessions" ?
But this seems only valid when the session provider is file and not for instance memory